### PR TITLE
Affected Issue(s): bugfix circular dependency if this artifact added as dependency in Spring Boot 2.6+ app

### PR DESCRIPTION
**Pull request by muhammad-levi**
What this commit has achieved:
1. Moved bean `keycloakConfigResolver` to a new `@Configuration` class,
to solve bug Spring Boot 2.6.x about circular dependency. (ref:
https://stackoverflow.com/questions/70207564/spring-boot-2-6-regression-how-can-i-fix-keycloak-circular-dependency-in-adapte)

### CODE REVIEW

File: src/main/java/autoconfigure/KeycloakSecurityConfig.java [hunk 1]
```diff
@@ -52,6 +51,7 @@
 
 @ConditionalOnProperty(prefix = "keycloak", name = "enabled", havingValue = "true", matchIfMissing = true)
 @KeycloakConfiguration
+@Import({ KeycloakConfigResolverConfig.class })
 public class KeycloakSecurityConfig extends KeycloakWebSecurityConfigurerAdapter {
```
Comment (muhammad-levi) [inline src/main/java/autoconfigure/KeycloakSecurityConfig.java:55]: The circular dependency injection is in the parent class.

image

**Comment by dubdabasoduba** [comment]
@muhammad-levi please give this PR an appropriate title

**Comment by muhammad-levi** [comment]
>

**Comment by rehammuzzamil** [comment]
PR looks good to me. But we also need to test it with the JPA server to make sure things work correctly. @dubdabasoduba

**Comment by rehammuzzamil** [comment]
@muhammad-levi  Please bump up version in pom.xml
cc : @dubdabasoduba

**Comment by muhammad-levi** [comment]
> @muhammad-levi Please bump up version in pom.xml cc : @dubdabasoduba

@rehammuzzamil Done 👌

**Pull request closed without merging**